File: Sorting/SortingNo1.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        // command 개수 변수로 저장
        int size = commands.length;
        int[] answer = new int[size];
        
        for(int i = 0 ; i < size ; i++){
            
            // i번째 부터 j번째 까지 자르고 ArrayList에 저장
            ArrayList<Integer> arr = new ArrayList<>();
            for(int j = commands[i][0]-1 ; j < commands[i][1] ; j++){
                arr.add(array[j]);
            }
            
            // 오름차순 정렬
            Collections.sort(arr);
            
            // k번째 추출 후 answer[]에 저장
            answer[i] = arr.get(commands[i][2]-1);
        }
        
        return answer;
    }
}

File: Sorting/SortingNo1.kt
// 풀이 1회차
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = IntArray(commands.size)
        
        var n = 0
        for(command in commands){
            var i = command[0]-1
            var j = command[1]
            var arr = ArrayList<Int>()
            while(i<j){
                arr.add(array[i])
                i++;
            }
            arr.sort()
            var k = command[2] - 1
            answer.set(n, arr.get(k))
            n++
        }
        
        return answer
    }
}

// kotlin 장점 살린 풀이 
class Solution {
        fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
            return commands.map { command ->
                array.slice(IntRange(command[0] - 1, command[1] - 1)).sorted()[command[2] - 1]
            }
                .toIntArray()
        }
    }